capture_service/dive_client_cli.cc

```diff
@@ -367,24 +458,26 @@ bool run_and_capture(Dive::DeviceManager& mgr,
                 gfxr_capture_directory,
                 gfxr_capture_frames,
                 is_gfxr_capture);
-    int time_to_wait_in_seconds = absl::GetFlag(FLAGS_trigger_capture_after);
-    std::cout << "wait for " << time_to_wait_in_seconds << " seconds" << std::endl;
-    std::this_thread::sleep_for(std::chrono::seconds(time_to_wait_in_seconds));
 
     if (is_gfxr_capture)
     {
+        trigger_gfxr_capture(mgr, package, gfxr_capture_frames);
         retrieve_gfxr_capture(mgr, gfxr_capture_directory);
     }
     else
     {
+        int time_to_wait_in_seconds = absl::GetFlag(FLAGS_trigger_capture_after);
+        std::cout << "wait for " << time_to_wait_in_seconds << " seconds" << std::endl;
+        std::this_thread::sleep_for(std::chrono::seconds(time_to_wait_in_seconds));
+
         trigger_capture(mgr);
-    }
 
-    std::cout << "Press Enter to exit" << std::endl;
-    std::string input;
-    if (std::getline(std::cin, input))
-    {
-        std::cout << "Exiting..." << std::endl;
+        std::cout << "Press Enter to exit" << std::endl;
```

Comment:
not really your code, but it seems that the user can press any key to exit? (not just Enter?)

Reply:
Yes, any other key would exit. Updated the message.